asllib/types.ml

```diff
@@ -774,9 +839,11 @@ let rec base_value loc env t =
       let e = normalize env e in
       E_Call ("Zeros", [ e ], []) |> add_pos_from t
   | T_Enum li -> IMap.find (List.hd li) env.global.constants_values |> lit
-  | T_Int None | T_Int (Some []) -> !$0
-  | T_Int (Some (Constraint_Exact e :: _))
-  | T_Int (Some (Constraint_Range (e, _) :: _)) ->
+  | T_Int UnConstrained | T_Int (UnderConstrained _) -> !$0
```

Comment:
How come the base value of an underconstrained integer is 0? I thought the LRM says there is no base value for underconstrained integers.

Reply:
An underconstrained integer has only one value it can take - the one given by the dynamic environment. So, either add a dynamic environment as another parameter or raise an exception.
In any case, we need to define the intended semantics.